zookeeper-server/src/main/java/org/apache/zookeeper/server/PrepRequestProcessor.java

```diff
@@ -628,6 +649,15 @@ private void pRequest2TxnCreate(int type, Request request, Record record, boolea
         } else {
             request.setTxn(new CreateTxn(path, data, listACL, createMode.isEphemeral(), newCversion));
         }
+        if (createMode.isEphemeral()) {
+            int count = this.ephemeralCount.getOrDefault(request.sessionId, 0);
+            int limit = Integer.getInteger("zookeeper.ephemeral.count.limit", 10000);
+            if (count >= limit) {
```

Comment:
we may need a logic here if `setting zookeeper.ephemeral.count.limit` to -1, users can turn off this limitation?